localstack/services/dynamodb/provider.py

```diff
@@ -498,37 +501,36 @@ def delete_table(self, context: RequestContext, table_name: TableName) -> Delete
 
         return result
 
-    def _forward_request(self, context: RequestContext, region: str | None) -> ServiceResponse:
-        """This helper is used to modify the request context for operations that make use of a global table region."""
-        if region:
-            with modify_context_region(context, region):
-                return self.forward_request(context)
-        return self.forward_request(context)
-
     def describe_table(self, context: RequestContext, table_name: TableName) -> DescribeTableOutput:
-        global_table_region = find_global_table_region(context, table_name)
-        # Check if table exists, to avoid error log output from DynamoDBLocal
-        if (
-            global_table_region
-            and not self.table_exists(context.account_id, global_table_region, table_name)
-        ) or not self.table_exists(context.account_id, context.region, table_name):
-            raise ResourceNotFoundException("Cannot do operations on a non-existent table")
+        global_table_region = self.get_global_table_region(context, table_name)
 
         result = self._forward_request(context=context, region=global_table_region)
 
-        # update response with additional props
+        # Update table properties from LocalStack stores
         table_props = get_store(context.account_id, context.region).table_properties.get(table_name)
         if table_props:
             result.get("Table", {}).update(table_props)
 
-        replicas: Dict = get_store(context.account_id, context.region).REPLICA_UPDATES.get(
-            table_name
-        )
-        if replicas and replicas.get(context.region):
-            regions = replicas.get(context.region)
-            result.get("Table", {}).update(
-                {"Replicas": [{"RegionName": region} for region in regions]}
+        # Update replication details
+        replicas: dict[str, set[str]] = get_store(
+            context.account_id, context.region
+        ).REPLICA_UPDATES.get(table_name, {})
+
+        replica_description_list = []
+        for source_region, replicated_regions in replicas.items():
+            # Contrary to AWS, we show all regions including the current context region where a replica exists
+            # This is due to the limitation of internal request forwarding mechanism for global tables
+            replica_description_list.append(
+                ReplicaDescription(RegionName=source_region, ReplicaStatus=ReplicaStatus.ACTIVE)
             )
+            for replicated_region in replicated_regions:
+                if replicated_region != context.region:
+                    replica_description_list.append(
+                        ReplicaDescription(
+                            RegionName=replicated_region, ReplicaStatus=ReplicaStatus.ACTIVE
+                        )
+                    )
+        result.get("Table", {}).update({"Replicas": replica_description_list})
```

Comment:
This makes sure that replica info is included when querying both the original table and replicated tables.